connexion/app.py

```diff
@@ -134,7 +134,12 @@ def add_api(self, swagger_file, base_path=None, arguments=None, auth_all_paths=N
         # TODO test if base_url starts with an / (if not none)
         arguments = arguments or dict()
         arguments = dict(self.arguments, **arguments)  # copy global arguments and update with api specfic
-        yaml_path = self.specification_dir / swagger_file
+
+        if isinstance(swagger_file, dict):
```

Comment:
We should change this name to `specification` to be generic because we don't know if the it is a file or the specification itself. The naming here needs to be reviewed, right now it is totally misleading.